### PR TITLE
added state only flag to fetch

### PR DESCRIPTION
**Pull request by roironn**

### CODE REVIEW

File: packages/cli/src/commands/fetch.ts [hunk 1]
```diff
@@ -352,6 +383,13 @@ const fetchBuilder = createCommandBuilder({
         default: false,
         demandOption: false,
       },
+      'state-only': {
+        alias: ['o'],
```
Comment (adamro) [inline packages/cli/src/commands/fetch.ts:387]: `o` sound sooo "output" :)
why not `s` btw?

File: packages/cli/src/commands/fetch.ts [hunk 2]
```diff
@@ -352,6 +383,13 @@ const fetchBuilder = createCommandBuilder({
         default: false,
         demandOption: false,
       },
+      'state-only': {
```
Comment (adamro) [inline packages/cli/src/commands/fetch.ts:386]: You need to change the `fetch` description. 
It's currently `Syncs this workspace\'s NaCl files with the services\' current state` which contradicts this option. 

Suggestion - `Syncs this workspace\'s with the services\' current state`

File: packages/cli/src/prompts.ts [hunk 3]
```diff
@@ -185,4 +187,14 @@ ${Prompts.SERVICE_ADD_HELP}`
   public static readonly APPROVE_ISOLATED_RECOMMENDATION = (
     newServicesNames: string
   ): string => `Would you like to switch to isolated mode and fetch ${newServicesNames}? (Answer No to continue without switching)`
+
+  public static readonly STATE_ONLY_UPDATE_START = (
```
Comment (adamro) [inline packages/cli/src/prompts.ts:191]: if you are only using the `.length` here and in `STATE_ONLY_UPDATE_FAILED` why are you passing `FetchChange[]` and not a number?

Also, the text for non-state is `Applying ${numChanges} changes to the local workspace` which is supposed to indicate the apply stage already stated and not going to start. This is the same so a similar phrasing makes sense I think.

Suggestion - `Applying ${numChanges} changes to the state. Workspace will not be updated.`

File: packages/cli/src/commands/fetch.ts [hunk 4]
```diff
@@ -247,34 +291,18 @@ export const fetchCommand = async (
...
-    : await getApprovedChanges(changes, interactive)
 
-  cliTelemetry.changesToApply(changesToApply.length, workspaceTags)
-  const updatingWsEmitter = new StepEmitter()
-  fetchProgress.emit('workspaceWillBeUpdated', updatingWsEmitter, changes.length, changesToApply.length)
-  const updatingWsSucceeded = await updateWorkspace(workspace, output, changesToApply, isolated)
+  const updatingWsSucceeded = stateOnly
+    ? await applyChangesToState(changes)
+    : await applyChangesToWorkspace(changes, isolated)
+  // If the workspace starts empty there is no point in showing a huge amount of changes
```
Comment (adamro) [inline packages/cli/src/commands/fetch.ts:298]: I think this comment be at line 207 now

File: packages/cli/src/workspace/workspace.ts [hunk 5]
```diff
@@ -142,6 +142,12 @@ export const loadWorkspace = async (workingDir: string, cliOutput: CliOutput,
   return { workspace, errored: status === 'Error' }
 }
 
+export const updateStateOnly = async (ws: Workspace,
+  changes: readonly FetchChange[]): Promise<void> => {
+  log.info('applying %d changes to state only', changes.length)
```
Comment (adamro) [inline packages/cli/src/workspace/workspace.ts:147]: we don't want to log the changes like we do in regular fetch?

File: packages/cli/test/commands/fetch.test.ts [hunk 6]
```diff
@@ -356,6 +361,34 @@ describe('fetch command', () => {
...
+              shouldUpdateConfig: mockUpdateConfig,
+              shouldCalcTotalSize: true,
+              approveIsolatedMode: mockApproveIsolatedModeTrue,
+              stateOnly: true,
+            })
+            expect(result).toBe(CliExitCode.Success)
+          })
+          it('should not apply any changes', () => {
+            const calls = findWsUpdateCalls(workspaceName)
+            expect(calls).toHaveLength(0)
```
Comment (adamro) [inline packages/cli/test/commands/fetch.test.ts:389]: maybe also check `ws.flush()` was actually called?

File: packages/cli/src/commands/fetch.ts [hunk 7]
```diff
@@ -189,6 +200,39 @@ export const fetchCommand = async (
...
+    }
+    updatingWsEmitter.emit('failed')
+    return false
+  }
+
+  const applyChangesToState = async (allChanges: readonly FetchChange[]): Promise<boolean> => {
+    const updatingStateEmitter = new StepEmitter()
+    fetchProgress.emit('stateWillBeUpdated', updatingStateEmitter, allChanges)
+    await updateStateOnly(workspace, allChanges)
+    updatingStateEmitter.emit('completed')
```
Comment (adamro) [inline packages/cli/src/commands/fetch.ts:232]: There's no failure scenario here? cause atm the `emit('failed')` will never be called for this emitter and thus the fail Prompt will never actually be used.

File: packages/cli/src/commands/fetch.ts (hunk 1 above)
Comment (roironn) [reply packages/cli/src/commands/fetch.ts:387]: Already taken ( specify the specific service to fetch )

Comment (adamro) [reply packages/cli/src/commands/fetch.ts:387]: right

maybe we need a new branding for this :)
`quiet` / `passive` / `weak` 

or, is it possible to have a 2 letters alias? cause 'st' works I guess..

Comment (roironn) [reply packages/cli/src/commands/fetch.ts:387]: That would make it `--st`. Kind of weird, but I'm ok with it. Thoughts?

File: packages/cli/src/commands/fetch.ts (hunk 2 above)
Comment (roironn) [reply packages/cli/src/commands/fetch.ts:386]: 👍

**Pull request merged**